File: tests/tests_old_compiler/ConditionalRemoteCalls/jt32.c
void remoteEchoReceiver(char *s, int count);
int processMessage(char *s);
int anotherProcessMsg(char *s);

jtask* localfuncA(int n, char *s) {
    int x, y;
    
    while(1) {
        printf("FuncA --- Indx %d message - %s\n", n, s);
        x = processMessage("Hello there from worker");
        y = anotherProcessMsg("Hi there from worker");
        printf("x = %d, y = %d\n", x, y);
        jsleep(500000);
    }
}

jtask* localfuncB(int n, char *s) {
    int x, y;

    while(1) {
        printf("FuncB -- Indx %d message - %s\n", n, s);
        x = processMessage("Hello there from worker");
        y = anotherProcessMsg("Hi there from worker");
        printf("x = %d, y = %d\n", x, y);
        jsleep(500000);
    }
}


int main(int argc, char *argv[])
{
    localfuncA(10, "a-local-task");
    localfuncB(20, "another-local-task");
    return 0;
}

File: tests/tests_old_compiler/ConditionalRemoteCalls/jt32.js

jcond {
    deviceOnly: jsys.type == "device";
    fogOnly: jsys.type == "fog";
}

let fcount = 10;
let scount = 10;
let count = 10;

jtask {fogOnly} function processMessage(str) {
    fcount++;
    console.log("First .. Processing the message ... ",  fcount, str);
    return fcount;
}

jtask* {fogOnly} function controllerFunc(str) {
    count++;
    console.log("Printing at the Controller... ", count, str);
}

jtask {deviceOnly} function anotherProcessMsg(str) {
    scount++;
    console.log("First .. Processing the message ... ",  scount, str);
    return scount;
}

setInterval(()=> {
    console.log("This is a loop at the controller.", jsys.tags);
    controllerFunc("hello, from controller").catch((e)=>{
        console.log("calling ControllerFunc at " + jsys.type + " has an Error: " + e);
    });
}, 1000);